### PR TITLE
Upgrade to async/await, tokio 0.2

### PR DESCRIPTION
**Pull request by ramn**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,10 +15,10 @@ keywords = ["metrics", "cloudwatch", "aws"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-chrono = "0.4.9"
-futures = "0.1.29"
+chrono = "0.4.10"
+futures = { version = "0.3.4", features = ["compat"] }
```
Comment (Marwes) [inline Cargo.toml]: Is compat needed if you are using rusutos beta?

File: src/collector.rs [hunk 2]
```diff
@@ -61,79 +61,79 @@ struct RecorderHandle(mpsc::Sender<Datum>);
...
 }
 
 pub fn init(config: Config) {
     let _ = thread::spawn(|| {
-        tokio::runtime::current_thread::run(init_future(config).map_err(|e| {
-            log::warn!("{}", e);
-        }));
+        let mut runtime = tokio::runtime::Builder::new()
+            // single threaded
+            .basic_scheduler()
```
Comment (Marwes) [inline src/collector.rs:71]: You may need to `enable_time` `enable_io` https://docs.rs/tokio/0.2.11/tokio/runtime/struct.Builder.html#method.enable_time

File: src/collector.rs [hunk 3]
```diff
@@ -61,79 +61,79 @@ struct RecorderHandle(mpsc::Sender<Datum>);
 
 struct Collector {
     metrics_data: BTreeMap<Timestamp, HashMap<(Key, Kind), StatisticSet>>,
-    config: Config,
+    config: &'static Config,
```
Comment (Marwes) [inline src/collector.rs:64]: Why change this to a reference?

File: Cargo.toml (hunk 1 above)
Comment (ramn) [reply Cargo.toml]: Good catch!

File: src/collector.rs (hunk 2 above)
Comment (ramn) [reply src/collector.rs:71]: Just discovered that.. thanks for catching it!

File: src/collector.rs (hunk 3 above)
Comment (ramn) [reply src/collector.rs:64]: To get rid of arbitrary cloning here and there - the Config object might as well be a static ref I thought.

Comment (Marwes) [reply src/collector.rs:64]: Ok, though it affected the public API, but it seems not to.

**Pull request merged**